pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -99,10 +101,26 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 
 	errs := []error{}
 	for _, node := range nodes {
-		if err := c.createSecretForNode(node, recorder); err != nil {
+		nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if err := c.checkSecrets(node, nodeInternalIPs, recorder); err != nil {
+			// if any of the secrets are not present, create all secrets for the node
+			if err := c.createSecretsForNode(node, nodeInternalIPs, recorder); err != nil {
```

Comment:
I believe you only want to create new secrets if `if errors.IsNotFound(err)` otherwise this appears to try and create a new certificate if there is any err even transient in `checkSecrets` ?  Is that true, what would be the result?